umaserverdemo/src/main/kotlin/com/lightspark/Vasp1.kt

```diff
@@ -273,6 +324,51 @@ class Vasp1(
         return "OK"
     }
 
+    private suspend fun handleNonUmaPayReq(call: ApplicationCall, nonUmaLnurlpResponse: NonUmaLnurlpResponse, amount: Long): String {
+        val url = URLBuilder(nonUmaLnurlpResponse.callback)
+        url.parameters.append("amount", amount.toString())
+        val urlWithAmount = url.buildString()
+
+        val response = httpClient.get(urlWithAmount)
+
+        if (response.status != HttpStatusCode.OK) {
+            call.respond(HttpStatusCode.InternalServerError, "Payreq to vasp2 failed: ${response.status}")
+            return "Payreq to vasp2 failed: ${response.status}"
+        }
+
+        val payReqResponse = try {
+            Json.decodeFromString<NonUmaPayReqResponse>(response.body())
+        } catch (e: Exception) {
+            call.respond(HttpStatusCode.InternalServerError, "Failed to parse payreq response.")
+            return "Failed to parse payreq response."
+        }
+
+        val invoice = try {
+            lightsparkClient.decodeInvoice(payReqResponse.pr)
+        } catch (e: Exception) {
+            call.respond(HttpStatusCode.InternalServerError, "Error decoding invoice.")
+            return "Error decoding invoice."
+        }
+
+        val newCallbackId = requestDataCache.savePayReqData(
+            encodedInvoice = payReqResponse.pr,
+            utxoCallback = getUtxoCallback(call, "1234abc"),
```

Comment:
This utxo callback should actually always be empty in the case of non-uma lnurl.